ironfish/src/utils/transaction.ts

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+const MAX_VERSION_DELTA = 10
+const MIN_VERSION_DELTA = 3
+
+/**
+ * Returns the sequence delta to use when checking which TransactionVersion to
+ * use for creating a new Transaction. Optimistically chooses a higher value to
+ * give the Transaction a better chance of being accepted when there is a
+ * version change happening soon.
+ */
+function versionSequenceDelta(expirationDelta?: number): number {
+  if (expirationDelta) {
+    // If the expirationDelta is less than the minimum, we must use that one
+    if (expirationDelta < MIN_VERSION_DELTA) {
+      return Math.max(1, expirationDelta)
```

Comment:
Wouldn't this be always `1`?

`expirationDelta` (if I understood correctly) is a non-negative integer. The `if (expirationDelta)` at the top ensures it's non-zero, hence when it gets here it's a positive integer. `max(1, positiveInteger)` should always evaluate to `1`. Am I missing something?

Reply:
I may be misunderstanding, but `max(1, positiveInteger)` should always evaluate to `positiveInteger`, technically. To be overly descriptive, it would be `1` instead of `positiveInteger` if `positiveInteger` is also `1`, but I think the point remains the same. `min(1, positiveInteger)` would indeed always evaluate to `1`, though.

It is _possible_ to call this function with a negative number, though, so I felt it better to be defensive here. `expirationDelta` _should_ be a positive integer, but technically there's no validation on `createTransaction`'s `expirationDelta` parameter enforcing a positive integer. The config setting `transactionExpirationDelta` does enforce positive integer.

Reply:
Right, sorry, I meant that it will always be `expirationDelta`. The core of the issue is that I don't fully understand what this method is supposed to do, and what its input should represent. Perhaps improving the comments and adding some assertions would be nice (e.g. if `expirationDelta` is supposed to be a positive integer, I'd add an assertion for that rather than using `max(1, expirationDelta)`--why would a negative `expirationDelta` be treated as 1 rather than 0?)